geoengine/__init__.py

```diff
@@ -11,10 +11,12 @@
 from .error import GeoEngineException, InputException, UninitializedException, TypeException, \
     MethodNotCalledOnPlotException, MethodNotCalledOnRasterException, MethodNotCalledOnVectorException, \
     SpatialReferenceMismatchException, check_response_for_error, ModificationNotOnLayerDbException, \
-    NoAdminSessionException, InvalidUrlException, MissingFieldInResponseException
+    InvalidUrlException, MissingFieldInResponseException
 from .layers import Layer, LayerCollection, LayerListing, LayerCollectionListing, \
     LayerId, LayerCollectionId, LayerProviderId, \
     layer_collection, layer
+from .permissions import add_permission, remove_permission, add_role, remove_role, assign_role, revoke_role, \
+    ADMIN_ROLE_ID, REGISTERED_USER_ROLE_ID, ANONYMOUS_USER_ROLE_ID, ResourceId, Permission, UserId, RoleId
```

Comment:
either remove the re-export or add them as real types. Would be nice to somehow get a ResourceId from a LayerId without writing the dict. Maybe add a from_layer_id(layer_id) to the new type. Or add an "ressource_id(self)" to LayerId

Reply:
cb77f6e92122f0cc346dbc4ab2320a44a0ceea55